File: cp_rust/src/bin/cses-1083-missing-number.rs
// Created by Ayush Biswas at 2025/07/09 22:11
// https://cses.fi/problemset/task/1083
use cp_lib::*;

// @code begin
use cpio::*;

sol! {
    fn solution(
        n: usize,
        a: [usize]
    ) -> usize {
        let observed = a.into_iter().fold(0, |acc, ai| acc ^ ai);
        let theoretical = (1..=n).fold(0, |acc, i| acc ^ i);
        observed ^ theoretical
    }
}

// @code end

File: cp_haskell/bin/cses-1083-missing-number.hs
-- Created by Ayush Biswas at 2026/03/03 15:37
-- https://cses.fi/problemset/task/1083

-- @code begin
import Data.Bits (Bits (xor))

main :: IO ()
main = do
  n :: Int <- read <$> getLine
  a :: [Int] <- map read . words <$> getLine
  let res = foldl1 xor a `xor` foldl1 xor [1 .. n]
  print res

-- @code end